include/PICA/pica_frag_config.hpp

```diff
@@ -220,12 +233,18 @@ namespace PICA {
 			setupTevStage(4);
 			setupTevStage(5);
 #undef setupTevStage
+
+			fogConfig.mode = (FogMode)Helpers::getBits<0, 3>(regs[InternalRegs::TexEnvUpdateBuffer]);
```

Comment:
FogConfig should only be updated if fog is enabled otherwise it'll generate redundant shaders based on the (unused) fog settings